src/WhereConditions/WhereHasConditionsDirective.php

```diff
@@ -52,17 +52,15 @@ public function handleBuilder($builder, $whereConditions)
             return $builder;
         }
 
-        return $builder->whereHas(
-            $this->getRelationName(),
-            function ($builder) use ($whereConditions): void {
-                // This extra nesting is required for the `OR` condition to work correctly.
-                $builder->whereNested(
-                    function ($builder) use ($whereConditions): void {
-                        $this->handleWhereConditions($builder, $whereConditions);
-                    }
-                );
-            }
-        );
+        return $this->handleWhereConditions($builder, [
```

Comment:
It should be possible to abstract this call more cleanly.

Reply:
At the risk of sounding like a total noob. What do you exactly mean by abstracting the call more cleanly? You mean creating a separate method that accept the model and `HAS` parameters, and using that instead? Or something else?

Reply:
You call this function in a way that emulates what happens when being passed a GraphQL input, so you have to reconstruct this array shape. In terms of the layers of abstractions, that is moving the wrong way.

What should happen instead is:
- there is one underlying method to call from PHP
- another method converts the inputs from GraphQL to call that method